src/Components/Aspire.OpenAI/AspireOpenAIClientBuilderChatClientExtensions.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Data.Common;
+using Aspire.OpenAI;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using OpenAI;
+
+namespace Microsoft.Extensions.Hosting;
+
+/// <summary>
+/// Provides extension methods for registering <see cref="IChatClient"/> as a singleton in the services provided by the <see cref="IHostApplicationBuilder"/>.
+/// </summary>
+public static class AspireOpenAIClientBuilderChatClientExtensions
+{
+    private const string DeploymentKey = "Deployment";
+    private const string ModelKey = "Model";
+
+    /// <summary>
+    /// Registers a singleton <see cref="IChatClient"/> in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">An <see cref="AspireOpenAIClientBuilder" />.</param>
+    /// <param name="deploymentName">Optionally specifies which model deployment to use. If not specified, a value will be taken from the connection string.</param>
+    /// <param name="disableOpenTelemetry">Optional. If <see langword="true"/>, skips registering open telemetry support in the <see cref="IChatClient"/> pipeline.</param>
+    /// <returns>A <see cref="ChatClientBuilder"/> that can be used to build a pipeline around the inner <see cref="IChatClient"/>.</returns>
+    public static ChatClientBuilder AddChatClient(
+        this AspireOpenAIClientBuilder builder,
+        string? deploymentName = null,
+        bool disableOpenTelemetry = false)
+    {
+        return builder.HostBuilder.Services.AddChatClient(
+            services => CreateInnerChatClient(services, builder, deploymentName, disableOpenTelemetry));
+    }
+
+    /// <summary>
+    /// Registers a keyed singleton <see cref="IChatClient"/> in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">An <see cref="AspireOpenAIClientBuilder" />.</param>
+    /// <param name="serviceKey">The service key with which the <see cref="IChatClient"/> will be registered.</param>
+    /// <param name="deploymentName">Optionally specifies which model deployment to use. If not specified, a value will be taken from the connection string.</param>
+    /// <param name="disableOpenTelemetry">Optional. If <see langword="true"/>, skips registering open telemetry support in the <see cref="IChatClient"/> pipeline.</param>
+    /// <returns>A <see cref="ChatClientBuilder"/> that can be used to build a pipeline around the inner <see cref="IChatClient"/>.</returns>
+    public static ChatClientBuilder AddKeyedChatClient(
+        this AspireOpenAIClientBuilder builder,
+        string serviceKey,
+        string? deploymentName = null,
+        bool disableOpenTelemetry = false)
```

Comment:
For consistency with other components this might be called `disableTracing`.
Though we usually don't have these in extension methods but in settings classes (configuration binding) with an extra lambda to configure by code.

Reply:
You can skip this argument if the method respects the setting from the higher level registration. I think it's fine to set it once for all clients.

Reply:
> You can skip this argument if the method respects the setting from the higher level registration

That is now done.